### PR TITLE
Validates if "return" in ConditionalReturnsOnNewline is a keyword

### PR DESCRIPTION
**Pull request by marcelofabri**
Fixes https://github.com/realm/SwiftLint/issues/784

I haven't found a better way to get a token's string than using `NSString.substringWithRange`, so I might be missing something.

### CODE REVIEW

**Comment by marcelofabri** [comment]
Forgot to add a changelog entry, will do that later

File: Source/SwiftLintFramework/Rules/ConditionalReturnsOnNewline.swift [hunk 1]
```diff
@@ -23,22 +23,43 @@ public struct ConditionalReturnsOnNewline: ConfigurationProviderRule, Rule, OptI
...
+            }
+
+            let nsString: NSString = file.contents
+            return !tokens.filter { token in
+                guard SyntaxKind(rawValue: token.type) == .Keyword else {
+                    return false
+                }
+
+                let tokenRange = NSRange(location: token.offset, length: token.length)
+                let tokenString = nsString.substringWithRange(tokenRange)
```
Comment (norio-nomura) [inline Source/SwiftLintFramework/Rules/ConditionalReturnsOnNewline.swift]: `SyntaxToken`'s `offset` and `length` are byte based numbers.
We need to use SourceKittenFramework's byte based API's e.g. `NSString.substringWithByteRange(start:length:)`.

File: Source/SwiftLintFramework/Rules/ConditionalReturnsOnNewline.swift [hunk 2]
```diff
@@ -23,22 +23,43 @@ public struct ConditionalReturnsOnNewline: ConfigurationProviderRule, Rule, OptI
...
     )
 
     public func validateFile(file: File) -> [StyleViolation] {
         let pattern = "(guard|if)[^\n]*return[^\n]\n*"
         let excludingKinds = SyntaxKind.commentAndStringKinds()
-        return file.matchPattern(pattern, excludingSyntaxKinds: excludingKinds).map {
+        return file.rangesAndTokensMatching(pattern).filter { range, tokens in
+            let kinds = tokens.flatMap { SyntaxKind(rawValue: $0.type) }
+
+            guard kinds.filter(excludingKinds.contains).isEmpty else {
```
Comment (norio-nomura) [inline Source/SwiftLintFramework/Rules/ConditionalReturnsOnNewline.swift]: This check caused false negative when match contains comment or string.
e.g. `"if true { return \"YES\" } else { return \"NO\" }"`.

**Comment by marcelofabri** [comment]
Thanks for the tips, @norio-nomura! I've updated the PR 🚀

**Comment by marcelofabri** [comment]
Done 💯

**Comment by norio-nomura** [comment]
Thanks! 🙏

**Comment by jpsim** [comment]
Nice!

**Pull request merged**